powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/annotation/PowerAuthAnnotationInterceptor.java

```diff
@@ -96,14 +120,103 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
                     );
                     request.setAttribute(PowerAuth.AUTHENTICATION_OBJECT, authentication);
                 } catch (PowerAuthAuthenticationException ex) {
-                    // silently ignore here and make sure authentication object is null
+                    // Silently ignore here and make sure authentication object is null
                     request.setAttribute(PowerAuth.AUTHENTICATION_OBJECT, null);
                 }
             }
 
+            // Resolve @PowerAuthEncryption annotation
+            if (powerAuthEncryptionAnnotation != null) {
```

Comment:
IMO we are missing the logic here that takes encryption parameters from **signature** header in case it is present, and from **encryption** header in case the signature header is not present...

Reply:
Can we handle this logic in a separate ticket? I didn't need such logic in current implementation, because:

- `Create activation` - uses `@PowerAuthEncryption` annotation which uses **encryption** header, there is no signature HTTP header -> no conflict of headers
- `Vault unlock` - uses only **signature** header directly because it needs to access signature data, **encryption** header is not used -> no conflict of headers
- `Create token` - uses only `PowerAuthApiAuthentication` object which is built using **signature** header, **encryption** header is not used   -> no conflict of headers

I understand why we need this logic, however I would like to implement it separately because current implementation doesn't require it and the use case is most likely for external users. It will be also easier to review this logic in a smaller pull request.

Reply:
Raised as https://github.com/wultra/powerauth-restful-integration/issues/84